### DISCUSSION TITLE
Initial items of ListElement not shown for nested object element

### DISCUSSION THREAD

**Discussion by mpopelo** [vueform/vueform#78] [Questions] [answered]
### Environment

- vue: ^3.2.31
- @vueform/vueform: ^1.5.4
- vite: ^4.5.0
- @vitejs/plugin-vue: ^4.4.0
- theme: Vueform

### Reproduction

[https://codesandbox.io/p/devbox/romantic-jang-nlknw5](https://codesandbox.io/p/devbox/romantic-jang-nlknw5?file=%2Fsrc%2Fcomponents%2FMyForm.vue&layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522cloymde1j00063b5xd37jsw3b%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522cloymde1i00033b5xrrmnzro5%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522cloymde1i00043b5xgd0faqy9%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522cloymde1i00053b5xhkvfvdac%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B60%252C40%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522cloymde1i00033b5xrrmnzro5%2522%253A%257B%2522id%2522%253A%2522cloymde1i00033b5xrrmnzro5%2522%252C%2522activeTabId%2522%253A%2522cloymlfo500hk3b5xhl0cn64u%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522cloymde1h00023b5xsdkcylkh%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252FREADME.md%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fsrc%252Fcomponents%252FMyForm.vue%2522%252C%2522id%2522%253A%2522cloymlfo500hk3b5xhl0cn64u%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fsrc%252Fmain.js%2522%252C%2522id%2522%253A%2522cloymohl702w73b5xd0kucn69%2522%252C%2522mode%2522%253A%2522temporary%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522id%2522%253A%2522clph7x7x4006y3b5xz68fafg2%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fsrc%252FApp.vue%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%257D%252C%2522cloymde1i00053b5xhkvfvdac%2522%253A%257B%2522id%2522%253A%2522cloymde1i00053b5xhkvfvdac%2522%252C%2522tabs%2522%253A%255B%257B%2522type%2522%253A%2522TASK_PORT%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522port%2522%253A5173%252C%2522id%2522%253A%2522cloymdkzf004l3b5xmije7zys%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522activeTabId%2522%253A%2522cloymdkzf004l3b5xmije7zys%2522%257D%252C%2522cloymde1i00043b5xgd0faqy9%2522%253A%257B%2522id%2522%253A%2522cloymde1i00043b5xgd0faqy9%2522%252C%2522activeTabId%2522%253A%2522cloymdj4a003n3b5xyzp42cso%2522%252C%2522tabs%2522%253A%255B%257B%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522id%2522%253A%2522cloymdj4a003n3b5xyzp42cso%2522%252C%2522mode%2522%253A%2522permanent%2522%257D%255D%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D)

### Describe the bug

The form allows me to display existing data, but does not allow me to check a radio button because the initially empty form is not displayed.

**Comment by adamberecz** [comment]
This is the expected behaviour, because you pass an empty `data` using `sync` which means the list should not have any items.

To have a single item in the list with unchecked radio:

```js
const data = {
  fields: [
    {
      type: {
        name: null,
      },
    },
  ],
};
```

**Comment by mpopelo** [comment]
Sure, could I then clarify if this is the expected behavior in the following case? We don't use sync and we receive data from the server, the structure of which does not match the structure of the form, and in this case, should the form not be displayed or will the data just not be filled in the form?

This is a very common situation when bad data comes from the server and in this case, the form is not displayed at all. The expected behavior for me is when the form is always displayed, and the data can be set or partially set in the form (if everything is fine with them) or not set at all.

[https://codesandbox.io/p/devbox/adoring-williams-xg8m6d?file=%2Fsrc%2Fcomponents%2FMyForm.vue%3A41%2C16](https://codesandbox.io/p/devbox/adoring-williams-xg8m6d?file=%2Fsrc%2Fcomponents%2FMyForm.vue%3A2%2C43&layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522cloymde1j00063b5xd37jsw3b%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522cloymde1i00033b5xrrmnzro5%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522cloymde1i00043b5xgd0faqy9%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522cloymde1i00053b5xhkvfvdac%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B60%252C40%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522cloymde1i00033b5xrrmnzro5%2522%253A%257B%2522id%2522%253A%2522cloymde1i00033b5xrrmnzro5%2522%252C%2522activeTabId%2522%253A%2522cloymlfo500hk3b5xhl0cn64u%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522cloymde1h00023b5xsdkcylkh%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252FREADME.md%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fsrc%252Fcomponents%252FMyForm.vue%2522%252C%2522id%2522%253A%2522cloymlfo500hk3b5xhl0cn64u%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522state%2522%253A%2522IDLE%2522%252C%2522initialSelections%2522%253A%255B%257B%2522startLineNumber%2522%253A2%252C%2522startColumn%2522%253A43%252C%2522endLineNumber%2522%253A2%252C%2522endColumn%2522%253A43%257D%255D%257D%252C%257B%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fsrc%252Fmain.js%2522%252C%2522id%2522%253A%2522cloymohl702w73b5xd0kucn69%2522%252C%2522mode%2522%253A%2522temporary%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522id%2522%253A%2522clph7x7x4006y3b5xz68fafg2%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fsrc%252FApp.vue%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%257D%252C%2522cloymde1i00053b5xhkvfvdac%2522%253A%257B%2522id%2522%253A%2522cloymde1i00053b5xhkvfvdac%2522%252C%2522tabs%2522%253A%255B%257B%2522type%2522%253A%2522TASK_PORT%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522port%2522%253A5173%252C%2522id%2522%253A%2522cloymdkzf004l3b5xmije7zys%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522activeTabId%2522%253A%2522cloymdkzf004l3b5xmije7zys%2522%257D%252C%2522cloymde1i00043b5xgd0faqy9%2522%253A%257B%2522id%2522%253A%2522cloymde1i00043b5xgd0faqy9%2522%252C%2522activeTabId%2522%253A%2522cloymdj4a003n3b5xyzp42cso%2522%252C%2522tabs%2522%253A%255B%257B%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522id%2522%253A%2522cloymdj4a003n3b5xyzp42cso%2522%252C%2522mode%2522%253A%2522permanent%2522%257D%255D%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D)

**Reply by adamberecz** [reply, marked as answer]
It's only the data that doesn't get filled. If you don't use `sync` it will not even fill in any data from `v-model`. `sync` instructs Vueform to look for initial state and changes in `v-model`.

The form is in fact displayed in the first example you sent (you can see the label for "List") it's just that the list does not have any items (value is non-existent which translates to an empty array). That's why the checkboxes don't appear, because they're what the list element should repeat.

Here's an alternative of what you can do:
https://codesandbox.io/p/devbox/goofy-thompson-7zpjty

- no `sync`
- `v-model` uses `data`
- you load the initial state with `form.load()` (`data` will still receive updates)
- you set `data.fields` if missing in `formatLoad()`

The [`formatLoad`](https://vueform.com/reference/vueform#option-format-load) runs when you use `form.load(data, true)`. It helps you convert incoming data to the form's data structure.

Could this be what you needed?

**Reply by mpopelo** [reply]
Thank you very much, this solves my current problem. I still have some doubts, I'll see how often I'll need to resort to this solution in the future.

**Reply by adamberecz** [reply]
Let's see what comes up or if there's anything we can improve here.
